packages/spanner/system-test/spanner.js

```diff
@@ -976,17 +976,18 @@ var spanner = new Spanner(env);
       var id = generateName('id');
       var name = generateName('name');
 
-      return database.runTransaction()
-        .then(function(data) {
-          var transaction = data[0];
-
-          transaction.insert('Singers', {
-            SingerId: id,
-            Name: name
-          });
+      return database.runTransaction(function(err, transaction) {
+        if (err) {
+          return Promise.reject(err);
+        }
 
-          return transaction.commit();
+        transaction.insert('Singers', {
+          SingerId: id,
+          Name: name
         });
+
+        return transaction.commit();
```

Comment:
Our testing framework allows you to return a promise and if the promise is rejected the test will fail. Essentially here we're just checking to see if we can run/commit a transaction via promises.